analyzers/src/SonarAnalyzer.CSharp/Rules/MethodOverloadsShouldBeGrouped.cs

```diff
@@ -39,25 +39,18 @@ public sealed class MethodOverloadsShouldBeGrouped : MethodOverloadsShouldBeGrou
             SyntaxKind.ClassDeclaration,
             SyntaxKind.InterfaceDeclaration,
             SyntaxKind.StructDeclaration,
-            SyntaxKindEx.RecordClassDeclaration
+            SyntaxKindEx.RecordClassDeclaration,
+            SyntaxKindEx.RecordStructDeclaration,
         };
 
-        protected override MemberInfo CreateMemberInfo(SyntaxNodeAnalysisContext c, MemberDeclarationSyntax member)
-        {
-            if (!IsValidMemberForOverload(member))
-            {
-                return null;
-            }
-            if (member is ConstructorDeclarationSyntax constructor)
-            {
-                return new MemberInfo(c, member, constructor.Identifier, constructor.IsStatic(), false, true);
-            }
-            else if (member is MethodDeclarationSyntax method)
+        protected override MemberInfo CreateMemberInfo(SyntaxNodeAnalysisContext c, MemberDeclarationSyntax member) =>
+            member switch
             {
-                return new MemberInfo(c, member, method.Identifier, method.IsStatic(), method.Modifiers.Any(SyntaxKind.AbstractKeyword), true);
-            }
-            return null;
-        }
+                { } when !IsValidMemberForOverload(member) => null,
```

Comment:
This row tests for `null` and `IsValidMemberForOverload` also tests for `null` and also checks type for `(member as MethodDeclarationSyntax)`. 
We can simplify that by moving the type check here. It will also improve overall coverage and make it more readable.

Reply:
The `IsValidMemberForOverload` method is checking for explicit interface implementations and is not covered by a test. The whole case can be removed without any tests failing. The rule is about grouping overloads together. The RSpec mentions explicit interface implementations:

```
As it is common practice to group method declarations by implemented interface, no issue will be raised for implicit and explicit interface implementations if grouped together with other members of that interface.
```

I don't fully understand what is meant by that sentence. Either way, one or more tests with explicit interface members should be added.

Reply:
This method order should raise
```
Go(int)
Stop()
Prepare()
Go() // Noncompliant
```
while this method order should not
```
Go(int)
Stop()
Prepare()
Go() 
```
if there exists an interface for `Prepare()` and `Go() `, because they implement an interface (implicit or explicit) and are grouped logically by the interface. So they should not be ordered by name in such case.


Please keep the code (if possible :) ) and add UTs.